src/raw/mod.rs

```diff
@@ -1015,6 +1015,24 @@ impl<T> RawTable<T> {
         }
     }
 
+    /// Returns an iterator which consumes all elements from the table.
+    ///
+    /// Iteration starts at the provided iterator's current location.
+    ///
+    /// This method panics if the given iterator does not cover all items remaining in the table.
```

Comment:
Use a `debug_assert` instead. This method is unsafe anyways.

Reply:
This comment is incorrect now that the implementation has changed.

Maybe a separate parameter should be added to pass the number of items remaining since it may not necessarily match the item count in the map.

Reply:
I'm not sure I follow --- I think the comment is _now_ correct whereas it wasn't before? Previously this returned an `Option`, which I removed in favor of just the `debug_assert`. Or do you mean that this isn't true in release mode? I could make it a regular `assert!` if you'd prefer?

I don't think passing in the number of items makes sense here — if the number of items left in the iterator doesn't match the number of items left in the table, then this is pretty much guaranteed to do weird things (most likely leak memory).

Reply:
You should just update the comment to say that making sure the number of items matches is a pre-condition of calling the function. The current comment implies that the function will check it for you, which is false (in release mode).

Reply:
How's that?